### PR TITLE
add laminas coding standard

### PR DESCRIPTION
**Pull request by ghostwriter**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

- Remove `doctrine/coding-standard`
- Require `laminas/laminas-coding-standard`
- Add composer scripts
- Update composer.lock
- Update phpcs.xml.dist
- Update .gitignore
- Fix CS issues
- Fix Psalm issues - 100% typed

Supersedes #150 (updated base branch)

### CODE REVIEW

File: src/Changelog/BumpAndCommitChangelogVersionInterface.php [hunk 1]
```diff
@@ -8,7 +8,7 @@
 use Laminas\AutomaticReleases\Git\Value\SemVerVersion;
 use Laminas\AutomaticReleases\Gpg\SecretKeyId;
 
-interface BumpAndCommitChangelogVersion
+interface BumpAndCommitChangelogVersionInterface
```
Comment (Ocramius) [inline src/Changelog/BumpAndCommitChangelogVersionInterface.php:11]: Ooof, no, this is a net loss.

`laminas/laminas-coding-standard` is really a no-go, if it enforces this stuff.

**Review by Ocramius — CHANGES_REQUESTED** [review]
Clear no from me here: I'd rather stick to `doctrine/coding-standard`, which is much more precise and strict.

**Comment by froschdesign** [comment]
@Ocramius 
The intention of @ghostwriter was right: a package of Laminas and the coding standard of Laminas. And the contributing guidelines also [include this coding standard](https://github.com/laminas/.github/blob/2da6c0a8e2e255d105a62808686da260ef27de24/CONTRIBUTING.md).

**Comment by Ocramius** [comment]
Yes, but as a maintainer of this package, the intent, while well-intended, is ill-advised.

This package is fine with the current coding standard.

**Pull request closed without merging**